borg/archiver.py

```diff
@@ -1213,6 +1217,18 @@ def build_parser(self, args=None, prog=None):
           required).
         - The archive checks can be time consuming, they can be skipped using the
           --repository-only option.
+
+        The --verify-data option will perform a full integrity verification (as opposed to
+        checking the CRC32 of the segment) of data, which means reading the data from the
+        repository, decrypting and decompressing it. This is a cryptographically secure
+        verification for encrypted repositories and will detect any manipulation of the
+        data with a probability of ~1.
```

Comment:
this leaves somehow the question open what happens with unencrypted repos / what the assurance is for them.